### PR TITLE
Fix #11052: use latest varargs pattern syntax

### LINKED ISSUE #11052: A locally defined scala.Enumeration blows up at runtime
## Minimized code

```scala
class Foo {
  implicit class StringShould(private val s: String) {
    def should(right: => Unit) = right
  }

  "Foo" should {
    object Weekdays extends Enumeration {
      val Mon, Tue, Wed, Thu, Fri, Sat, Sun = Value
    }
    println(Weekdays.Mon.toString)
  }
}

object Main {
  def main(args: Array[String]): Unit = new Foo()
}
```

## Output

```
$ scala3 Foo.scala
Exception in thread "main" java.lang.IllegalAccessException: class scala.Enumeration cannot access a member of class Foo$Weekdays$1$ with modifiers "public"
	at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:361)
	at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:591)
	at java.base/java.lang.reflect.Method.invoke(Method.java:558)
	at scala.Enumeration.$anonfun$populateNameMap$3(Enumeration.scala:208)
	at scala.Enumeration.populateNameMap(Enumeration.scala:205)
	at scala.Enumeration.$anonfun$nameOf$1(Enumeration.scala:220)
	at scala.collection.mutable.HashMap.getOrElse(HashMap.scala:436)
	at scala.Enumeration.scala$Enumeration$$nameOf(Enumeration.scala:220)
	at scala.Enumeration$Val.toString(Enumeration.scala:263)
	at Foo.$init$$$anonfun$1(Foo.scala:10)
	at Foo$StringShould.should(Foo.scala:3)
	at Foo.<init>(Foo.scala:11)
	at Main$.main(Foo.scala:15)
	at Main.main(Foo.scala)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at dotty.tools.scripting.ScriptingDriver.compileAndRun(ScriptingDriver.scala:29)
	at dotty.tools.scripting.Main$.main(Main.scala:18)
	at dotty.tools.scripting.Main.main(Main.scala)
```

## Expectation

No runtime exception.

### LINKED ISSUE #11502: Old syntax is used in warning about vararg splice in pattern position
## Compiler version

3.0.0-RC1

## Minimized code

```Scala
List(1, 2, 3) match
    case x :: y :: Nil => ()
    case Nil => ()
```

## Output

~~~ text
match may not be exhaustive.

It would fail on pattern case: List(_, _, _, _: _*), List(_)
~~~

## Expectation

Warning should be the following?

~~~ text
match may not be exhaustive.

It would fail on pattern case: List(_, _, _, _ *), List(_)
~~~

### PR DESCRIPTION
**Pull request by liufengyun**
Fix #11502: use latest varargs pattern syntax

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/transform/patmat/Space.scala [hunk 1]
```diff
@@ -762,9 +762,9 @@ class SpaceEngine(using Context) extends SpaceLogic {
         if (ctx.definitions.isTupleType(tp))
           params(tp).map(_ => "_").mkString("(", ", ", ")")
         else if (scalaListType.isRef(sym))
-          if (flattenList) "_: _*" else "_: List"
+          if (flattenList) "_*" else "_: List"
         else if (scalaConsType.isRef(sym))
```
Comment (julienrf) [inline compiler/src/dotty/tools/dotc/transform/patmat/Space.scala:766]: Isn't '_*' parsed as a single token? Should it be '_ *' instead?

Comment (liufengyun) [reply compiler/src/dotty/tools/dotc/transform/patmat/Space.scala:766]: From the syntax specification, `_*` will be parsed correctly, and the implementation agrees with the syntax: https://scastie.scala-lang.org/EvlND8SaTMGnJjyhaqdRIA

Comment (julienrf) [reply compiler/src/dotty/tools/dotc/transform/patmat/Space.scala:766]: This snippet crashes the compiler.

Comment (liufengyun) [reply compiler/src/dotty/tools/dotc/transform/patmat/Space.scala:766]: I enabled `-Xprint:typer` in the above, the output is noisy. Here is a clean one: https://scastie.scala-lang.org/KnwXPu1kTQKYp9EGI8qrQw

**Review by julienrf — APPROVED** [review]
Thank you!

**Pull request merged**